Association_Mining/blog.md

```diff
@@ -0,0 +1,219 @@
+```
+
+We can create baskets for each country by executing the following code:
+
+```
+basket_USA = (data[data['Country'] =="USA"] 
+          .groupby(['Invoice', 'Description'])['Quantity'] 
+          .sum().unstack().reset_index().fillna(0) 
+          .set_index('Invoice'))
+```
+
+Here, we have filtered the data according to the country. We need to group the data according to *Invoice* number and *Description* and structure the data such that for each invoice, we can give a value *(Quantity)* to each item based on whether it is bought. If a certain item is bought in a transaction, we put the *quantity* or else a *0*. 
+
+Because, we are not concerned with the Quantity of the product sold, we convert the values to *0* or *1* indicating *False* or *True* respectively. It shows if the product is sold in a particular transaction.
+
+```
+def hot_encode(x): 
+    if(x<= 0): 
+        return 0
+    if(x>= 1): 
+        return 1
+        
+basket_encoded = basket_USA.applymap(hot_encode) 
+basket_USA = basket_encoded
+```
+
+Using the apriori function, and also assuming the support threshold to be *0.06*, we find the frequent item sets.
+
+```
+frq_items = apriori(basket_USA, min_support = 0.06, use_colnames = True)
+```
+
+To find the association rules, we use the *association_rules()* function. We pass the frequent item sets as one of the parameters. We use the **Lift** metric and set the minimum threshold to be 1.2.
+
+```
+rules = association_rules(frq_items, metric ="lift", min_threshold = 1.2)
+rules = rules.sort_values(['confidence', 'lift'], ascending =[False, False]) 
+```
+
+We also sort the rules based on the confidence value and the lift metric value such that the rules with higher confidence and lift are shown on the top.
+
+Let us see how the rules table looks. Execute the following code and you should get an output similar to the image shown below.
+
+```
+rules.head(6)
+```
+
+![rules_head output](/Association_Mining/images/rules_head.png)
+
+For example, here it shows that if a customer in USA buys a *Mini Paint Set Vintage*, he/she will most likely buy *60 Cake Cases Dolly Girl Design*.
+
+The support value for the first rule is 0.1. This number is calculated by dividing the number of transactions containing *Mini Paint Set Vintage* divided by total number of transactions. The confidence level for the rule is 1.0 which shows that out of all the transactions that contain *Mini Paint Set Vintage*, 100% of the transactions also contain *60 Cake Cases Dolly Girl Design*. Finally, the lift of 10.0 tells us that *60 Cake Cases Dolly Girl Design* is 10 times more likely to be bought by the customers who buy *Mini Paint Set Vintage* compared to the default likelihood of the sale of *60 Cake Cases Dolly Girl Design*.
+
+You can try this for other countries and find interesting association rules while changing the threshold values as required.
+
+## Conclusion
+
+* Audiences are able to understand the basic concept of association mining

```

Comment:
can you make this whole section in a paragraph style rather than bullets? And remember you are addressing the audience directly, so you don't need to say "audience should..", just say "we learned.." or "you should know how to.."